File: erlang/recursion/fibonacci.erl
-module(fibonacci).
-export([fibonacci/1, main/0]).

fibonacci(0) -> 0;
fibonacci(1) -> 1;
fibonacci(N) when N > 1 -> fibonacci(N-1) + fibonacci(N-2).

main() ->
    Res = fibonacci(45),
    io:format("Erlang result : ~p~n", [Res]).

File: rust/recursion/fibonacci.rs
fn fibonacci(n: u64) -> u64
{
    match n
    {
        0 => 0,
        1 => 1,
        _ => fibonacci(n - 1) + fibonacci(n - 2),
    }
}

fn main()
{
    let res = fibonacci(45);
    println!("Rust result : {}", res);
}